sdk/core/azure-core/src/exception.cpp

```diff
@@ -28,4 +30,48 @@ namespace Azure { namespace Core {
     RawResponse = std::move(rawResponse);
   }
 
+  RequestFailedException::RequestFailedException(
+      std::unique_ptr<Azure::Core::Http::RawResponse>& rawResponse)
+      : std::runtime_error(getRawResponseField(rawResponse, "message"))
+  {
+    auto& headers = rawResponse->GetHeaders();
```

Comment:
This method isn't cheap to call, and we are doing it 3 times per exception ctor. Maybe it is OK since this is the exception code path, but might be worth trying to avoid the unnecessary work.

Reply:
it's taking a ref to the headers map, why is it expensive?

Reply:
The implementation of `GetHeaders()` itself is expensive, calls `MergeMap`.
We were calling `GeHeaders()` three times here.